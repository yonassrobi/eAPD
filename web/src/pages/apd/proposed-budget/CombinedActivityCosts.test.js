import { shallow } from 'enzyme';
import React from 'react';
import {
  plain as CombinedActivityCosts,
  mapStateToProps,
  DataRow,
  DataRowGroup,
  SummaryBudgetByActivityTotals,
  SummaryBudgetByActivityBreakdown
} from './CombinedActivityCosts';

describe('renders correctly', () => {
  it('renders correctly in view-only mode', () => {
    const component = shallow(
      <CombinedActivityCosts data={{}} years={['1984', '1985']} isViewOnly />
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly in standard mode', () => {
    const component = shallow(
      <CombinedActivityCosts data={{}} years={['1984', '1985']} />
    );
    expect(component).toMatchSnapshot();
  });

  test('data row renders', () => {
    const component = shallow(
      <DataRow
        category="category"
        data={{ federal: 1, state: 2, medicaid: 1000, total: 3 }}
        title="title text"
        groupTitle="group title text"
      />
    );
    expect(component).toMatchSnapshot();
  });

  test('data row group renders', () => {
    expect(
      shallow(
        <DataRowGroup
          year="1448"
          data={{
            combined: { 1448: {} },
            contractors: { 1448: {} },
            expenses: { 1448: {} },
            statePersonnel: { 1448: {} }
          }}
          groupTitle="MMIS"
        />
      )
    ).toMatchSnapshot();
  });

  test('Combined Activity Costs Totals renders', () => {
    expect(
      shallow(
        <SummaryBudgetByActivityTotals
          data={{
            hit: {
              1975: { federal: 1, state: 2, medicaid: 1000, total: 3 },
              1976: { federal: 10, state: 20, medicaid: 2000, total: 30 }
            },
            hie: {
              1975: { federal: 2, state: 3, medicaid: 2000, total: 5 },
              1976: { federal: 20, state: 30, medicaid: 3000, total: 50 }
            },
            mmis: {
              1975: { federal: 3, state: 4, medicaid: 3000, total: 7 },
              1976: { federal: 30, state: 40, medicaid: 4000, total: 70 }
            },
            combined: {
              1975: { federal: 4, state: 5, medicaid: 4000, total: 9 },
              1976: { federal: 40, state: 50, medicaid: 5000, total: 90 }
            }
          }}
          ffy="1975"
        />
      )
    ).toMatchSnapshot();
  });

  test('Combined Activity Costs breakdown renders', () => {
    const data = {
      activityTotals: [
        {
          data: {
            otherFunding: {
              1990: {
                statePersonnel: 100,
                expenses: 200,
                contractors: 350,
                total: 650
              },
              1991: {
                statePersonnel: 400,
                expenses: 700,
                contractors: 150,
                total: 1250
              }
            }
          }
        },
        {
          data: {
            otherFunding: {
              1990: {
                statePersonnel: 200,
                expenses: 400,
                contractors: 150,
                total: 750
              },
              1991: {
                statePersonnel: 600,
                expenses: 1000,
                contractors: 250,
                total: 1850
              }
            }
          }
        }
      ]
    };
    expect(
      shallow(<SummaryBudgetByActivityBreakdown data={data} ffy="1975" />)
    ).toMatchSnapshot();
  });

  test('maps state to props', () => {
    const state = {
      apd: { data: { years: ['1', '2'] } },
      budget: {
        activityTotals: [
          { fundingSource: 'HIE', name: 'activity 1' },
          { fundingSource: 'HIT', name: 'activity 2' },
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      }
    };

    expect(mapStateToProps(state)).toEqual({
      data: {
        activityTotals: [
          { fundingSource: 'HIE', name: 'activity 1' },
          { fundingSource: 'HIT', name: 'activity 2' },
          { fundingSource: 'MMIS', name: 'activity 3' },
          { fundingSource: 'MMIS', name: 'activity 4' }
        ]
      },
      years: ['1', '2']
    });
  });
});
