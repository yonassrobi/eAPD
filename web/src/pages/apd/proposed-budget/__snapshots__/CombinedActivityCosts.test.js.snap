// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Combined Activity Costs Totals renders 1`] = `
<table
  className="budget-table"
  data-cy="CACTable"
>
  <thead>
    <tr
      className="budget-table--row__primary-header"
    >
      <th
        scope="col"
      >
        Combined Activity Costs FFY 
        1975
         (Total Computable Medicaid Cost)
      </th>
      <th
        className="ds-u-text-align--right"
        scope="col"
      >
        Total
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="budget-table--row__header"
    >
      <th
        colSpan="2"
        scope="row"
      >
        HIT
      </th>
    </tr>
    <DataRowGroup
      data={
        Object {
          "1975": Object {
            "federal": 1,
            "medicaid": 1000,
            "state": 2,
            "total": 3,
          },
          "1976": Object {
            "federal": 10,
            "medicaid": 2000,
            "state": 20,
            "total": 30,
          },
        }
      }
      groupTitle="HIT"
      year="1975"
    />
    <tr
      className="budget-table--row__header"
    >
      <th
        colSpan="2"
        scope="row"
      >
        HIE
      </th>
    </tr>
    <DataRowGroup
      data={
        Object {
          "1975": Object {
            "federal": 2,
            "medicaid": 2000,
            "state": 3,
            "total": 5,
          },
          "1976": Object {
            "federal": 20,
            "medicaid": 3000,
            "state": 30,
            "total": 50,
          },
        }
      }
      groupTitle="HIE"
      year="1975"
    />
    <tr
      className="budget-table--row__header"
    >
      <th
        colSpan="2"
        scope="row"
      >
        MMIS
      </th>
    </tr>
    <DataRowGroup
      data={
        Object {
          "1975": Object {
            "federal": 3,
            "medicaid": 3000,
            "state": 4,
            "total": 7,
          },
          "1976": Object {
            "federal": 30,
            "medicaid": 4000,
            "state": 40,
            "total": 70,
          },
        }
      }
      groupTitle="MMIS"
      year="1975"
    />
    <tr
      className="budget-table--row__header"
      key="1975"
    >
      <th
        scope="row"
      >
        FFY 
        1975
         Total Computable Medicaid Cost
      </th>
      <td
        className="budget-table--number budget-table--total"
      >
        <Dollars>
          4000
        </Dollars>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`renders correctly Combined Activity Costs breakdown renders 1`] = `
Array [
  <Connect(SummaryActivityBreakdownTable)
    activityIndex={0}
    ffy="1975"
    key="1975-0"
    otherFunding={
      Object {
        "1990": Object {
          "contractors": 350,
          "expenses": 200,
          "statePersonnel": 100,
          "total": 650,
        },
        "1991": Object {
          "contractors": 150,
          "expenses": 700,
          "statePersonnel": 400,
          "total": 1250,
        },
      }
    }
  />,
  <Connect(SummaryActivityBreakdownTable)
    activityIndex={1}
    ffy="1975"
    key="1975-1"
    otherFunding={
      Object {
        "1990": Object {
          "contractors": 150,
          "expenses": 400,
          "statePersonnel": 200,
          "total": 750,
        },
        "1991": Object {
          "contractors": 250,
          "expenses": 1000,
          "statePersonnel": 600,
          "total": 1850,
        },
      }
    }
  />,
]
`;

exports[`renders correctly data row group renders 1`] = `
<Fragment>
  <DataRow
    category="statePersonnel"
    data={Object {}}
    groupTitle="MMIS"
    key="MMIS-statePersonnel-State Staff Total"
    title="State Staff Total"
  />
  <DataRow
    category="expenses"
    data={Object {}}
    groupTitle="MMIS"
    key="MMIS-expenses-Other State Expenses Total"
    title="Other State Expenses Total"
  />
  <DataRow
    category="contractors"
    data={Object {}}
    groupTitle="MMIS"
    key="MMIS-contractors-Private Contractor Total"
    title="Private Contractor Total"
  />
  <DataRow
    category="combined"
    data={Object {}}
    groupTitle="MMIS"
    key="MMIS-combined-Total"
    title="Total"
  />
</Fragment>
`;

exports[`renders correctly data row renders 1`] = `
<tr
  className=""
  key="category"
>
  <th
    className="title indent-title"
    scope="row"
  >
    title text
  </th>
  <td
    className="budget-table--number"
  >
    <Dollars>
      1000
    </Dollars>
  </td>
</tr>
`;

exports[`renders correctly renders correctly in standard mode 1`] = `
Array [
  <Fragment
    key="1984"
  >
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      FFY 
      1984
    </h4>
    <Instruction
      args={null}
      headingDisplay={
        Object {
          "className": "ds-h3",
          "level": "h3",
        }
      }
      labelFor={null}
      reverse={false}
      source="proposedBudget.combinedActivityCosts.totalMedicaidCost"
    />
    <SummaryBudgetByActivityTotals
      data={Object {}}
      ffy="1984"
    />
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      State and Contractor Cost Breakdown
    </h4>
    <Instruction
      args={null}
      headingDisplay={
        Object {
          "className": "ds-h3",
          "level": "h3",
        }
      }
      labelFor={null}
      reverse={false}
      source="proposedBudget.combinedActivityCosts.activityBreakdown"
    />
    <SummaryBudgetByActivityBreakdown
      data={Object {}}
      ffy="1984"
    />
  </Fragment>,
  <Fragment
    key="1985"
  >
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      FFY 
      1985
    </h4>
    <Instruction
      args={null}
      headingDisplay={
        Object {
          "className": "ds-h3",
          "level": "h3",
        }
      }
      labelFor={null}
      reverse={false}
      source="proposedBudget.combinedActivityCosts.totalMedicaidCost"
    />
    <SummaryBudgetByActivityTotals
      data={Object {}}
      ffy="1985"
    />
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      State and Contractor Cost Breakdown
    </h4>
    <Instruction
      args={null}
      headingDisplay={
        Object {
          "className": "ds-h3",
          "level": "h3",
        }
      }
      labelFor={null}
      reverse={false}
      source="proposedBudget.combinedActivityCosts.activityBreakdown"
    />
    <SummaryBudgetByActivityBreakdown
      data={Object {}}
      ffy="1985"
    />
  </Fragment>,
]
`;

exports[`renders correctly renders correctly in view-only mode 1`] = `
Array [
  <Fragment
    key="1984"
  >
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      FFY 
      1984
    </h4>
    <SummaryBudgetByActivityTotals
      data={Object {}}
      ffy="1984"
    />
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      State and Contractor Cost Breakdown
    </h4>
    <SummaryBudgetByActivityBreakdown
      data={Object {}}
      ffy="1984"
    />
  </Fragment>,
  <Fragment
    key="1985"
  >
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      FFY 
      1985
    </h4>
    <SummaryBudgetByActivityTotals
      data={Object {}}
      ffy="1985"
    />
    <h4
      aria-hidden="true"
      className="ds-h4"
    >
      State and Contractor Cost Breakdown
    </h4>
    <SummaryBudgetByActivityBreakdown
      data={Object {}}
      ffy="1985"
    />
  </Fragment>,
]
`;
